### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by Minoiski**
Кретович

### CODE REVIEW

**Comment by gost402** [comment]
Add app screenshot to PR description

**Comment by gost402** [comment]
Add `.class` files to `.gitignore`

File: Кретович/Lab1/src/Lab/Main.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import javax.swing.JComponent;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+import javax.swing.Timer;
+
+
```
Comment (gost402) [inline Кретович/Lab1/src/Lab/Main.java]: Unnecessary empty line

File: Кретович/Lab1/src/Lab/Main.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+public class Main extends JComponent implements ActionListener
+{
+
+    private int[]xPoints={200,250,400};
+    private int[]yPoints={200,114,250};
```
Comment (gost402) [inline Кретович/Lab1/src/Lab/Main.java]: spaces after types, commas, `=` operators. Fix them everywhere, not only here

File: Кретович/Lab1/src/Lab/Main.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    private int n=3;
+    private Color borderColor = Color.green;
+    private Color backgroundColor = Color.red;
+
```
Comment (gost402) [inline Кретович/Lab1/src/Lab/Main.java]: Do you want empty line here?

File: Кретович/Lab1/src/Lab/Main.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+import javax.swing.Timer;
+
+
+public class Main extends JComponent implements ActionListener
+{
+
+    private int[]xPoints={200,250,400};
```
Comment (gost402) [inline Кретович/Lab1/src/Lab/Main.java]: Move all this to consts if they aren't changed somewhere

File: Кретович/Lab1/src/Lab/Main.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+
+    private static final int WIDTH = 800;
+    private static final int HEIGHT = 600;
+    private double theta = 0;
+    private static final float BORDER = 3;
+
+    private Timer timer;
+
+    private final double ROTATE_ANGLE = Math.PI / 180;
```
Comment (gost402) [inline Кретович/Lab1/src/Lab/Main.java]: static

File: Кретович/Lab1/src/Lab/Main.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    private int n=3;
+    private Color borderColor = Color.green;
+    private Color backgroundColor = Color.red;
+
+    private MyTriangle myTriangle;
+
+
+    private static final int WIDTH = 800;
+    private static final int HEIGHT = 600;
+    private double theta = 0;
```
Comment (gost402) [inline Кретович/Lab1/src/Lab/Main.java]: group your vars somehow, this is surrounded by consts, you have final var below and so on

**Comment by Minoiski** [comment]
default

File: Кретович/Lab1/src/Lab/Main.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    {
+        repaint();
+    }
+
+    @Override
+    protected void paintComponent(Graphics g)
+    {
+        Graphics2D graphics2D = (Graphics2D) g;
+        graphics2D.setStroke(new BasicStroke(BORDER));
+        myTriangle = new MyTriangle(borderColor, backgroundColor,theta,xPoints,yPoints,n);
```
Comment (gost402) [inline Кретович/Lab1/src/Lab/Main.java]: You create new `MyTriangle` object on each `repaint` action. It's better to make this belongs to `Main` and then draw only changing `theta` inside Timer action, `paintComponent` shouldn't be responsible for `angle` change

File: Кретович/Lab1/src/Lab/MyTriangle.java [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    public void paint(Graphics graphics)
+    {
+        Graphics2D graphics2D = (Graphics2D) graphics;
+        AffineTransform affineTransform = AffineTransform.getRotateInstance(this.theta,this.centerX,this.centerY);
+        graphics2D.setTransform(affineTransform);
+        graphics2D.setColor(backgroundColor);
+        graphics2D.fillPolygon(xPoints,yPoints,n);
+        graphics2D.setColor(borderColor);
+        graphics2D.drawPolygon(xPoints,yPoints,n);
+
```
Comment (gost402) [inline Кретович/Lab1/src/Lab/MyTriangle.java]: empty line

File: Кретович/Lab1/src/Lab/MyTriangle.java [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+        this.yPoints=yPoints;
+        this.n=numberOfPoints;
+        this.theta=theta;
+        backgroundColor=clr2;
+        borderColor=clr1;
+        this.centerX = (xPoints[0]+xPoints[1]+xPoints[2])/3;
+        this.centerY = (yPoints[0]+yPoints[1]+yPoints[2])/3;
+    }
+    @Override
+
```
Comment (gost402) [inline Кретович/Lab1/src/Lab/MyTriangle.java]: empty line, fix all of them

File: Кретович/Lab1/src/Lab/MyTriangle.java [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+        this.xPoints=xPoints;
+        this.yPoints=yPoints;
+        this.n=numberOfPoints;
+        this.theta=theta;
+        backgroundColor=clr2;
+        borderColor=clr1;
+        this.centerX = (xPoints[0]+xPoints[1]+xPoints[2])/3;
+        this.centerY = (yPoints[0]+yPoints[1]+yPoints[2])/3;
+    }
+    @Override
```
Comment (gost402) [inline Кретович/Lab1/src/Lab/MyTriangle.java]: Here SHOULD be empty line, after constructor body

File: Кретович/Lab1/src/Lab/MyTriangle.java [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+
+    private int[]xPoints;
+    private int[]yPoints;
+    private int n;
+    private int centerX;
+    private int centerY;
+    private Color borderColor;
+    private Color backgroundColor;
+    private double theta;
+    public MyTriangle(Color clr1,Color clr2,double theta,int []xPoints,int []yPoints,int numberOfPoints)
```
Comment (gost402) [inline Кретович/Lab1/src/Lab/MyTriangle.java]: add empty line after vars before constructor

Comment (gost402) [reply Кретович/Lab1/src/Lab/MyTriangle.java]: also commas...

File: Кретович/Lab1/УП 2 Курс.iml [hunk 12]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (gost402) [inline Кретович/Lab1/УП 2 Курс.iml]: To gitignore

**Pull request merged**